test/benchmark/bm_proto_creation.cpp

```diff
@@ -97,64 +97,107 @@ void checkLoop(const T &obj) {
   }
 }
 
-BENCHMARK_F(BlockBenchmark, CopyTest)(benchmark::State &st) {
-  auto block = complete_builder.build();
+/**
+ * Runs a function and updates timer of the given state
+ */
+template <typename Func>
+void runBenchmark(benchmark::State &st, Func &&f) {
+  auto start = std::chrono::high_resolution_clock::now();
+  f();
+  auto end   = std::chrono::high_resolution_clock::now();
 
+  auto elapsed_seconds =
+      std::chrono::duration_cast<std::chrono::duration<double>>(
+          end - start);
+
+  st.SetIterationTime(elapsed_seconds.count());
+}
+
+BENCHMARK_DEFINE_F(BlockBenchmark, TransportCopyTest)(benchmark::State &st) {
```

Comment:
Add documentation about a semantic of the performance test. Also, this issue is related to tests below.